### PR TITLE
`Xml::build()` can handle urls and json data

### PR DESCRIPTION
**Pull request by mirko-pagliai**
Why the `Xml` utility should not handle directly urls and json data?

Note: for now I have only entered the code in this PR, to know whether if you like it.  
We will need to update the method description and documentation, in the case.

### CODE REVIEW

File: src/Utility/Xml.php [hunk 1]
```diff
@@ -108,7 +109,17 @@ public static function build($input, array $options = [])
             'readFile' => true
         ];
         $options += $defaults;
-
+        
+        if (!filter_var($input, FILTER_VALIDATE_URL) === false) {
```
Comment (dereuromark) [inline src/Utility/Xml.php]: if (!... === false) looks like a double negation.

**Comment by ionas** [comment]
Why make this depend on each other when you can in your user land app simply combine it?

File: src/Utility/Xml.php (hunk 1 above)
Comment (mirko-pagliai) [inline src/Utility/Xml.php]: You're right. But, for now, I just want to know if you like it.

**Comment by dereuromark** [comment]
I agree with @ionas, this seems to add a snowflake feature inside the Xml class.

**Comment by mirko-pagliai** [comment]
@dereuromark what means "snowflake feature"?

File: src/Utility/Xml.php [hunk 2]
```diff
@@ -108,7 +109,17 @@ public static function build($input, array $options = [])
...
+            $http = new Client();
+            $response = $http->get($input);
+            $input = $response->body();
+        }
```
Comment (markstory) [inline src/Utility/Xml.php]: This is a huge security issue. That's why we removed this feature. Imagine a scenario where you are accepting an XML string from a user. They can now have your server do a request to their misbehaving server and easily perform a denial of service attack.

**Comment by markstory** [comment]
I can't merge this. It is too dangerous.

**Comment by mirko-pagliai** [comment]
Perfectly clear. Thank you!

**Pull request closed without merging**